tests/test_ciffilereaderwriter.py

```diff
@@ -162,6 +165,16 @@ def test_cif_read_write(self):
             # This should fail since it's using 2d positions
             traj[-1].cif_coordinates = [[0, 0], [0, 0]]
 
+    def test_hexagonal(self):
+        with open(HP2_PATH, 'r') as f:
+            default_trajectory = self.read_cif_trajectory(f)
+
+            f.seek(0)
+            bad_trajectory = self.read_cif_trajectory(f, tolerance=1e-5)
+
+            self.assertEqual(len(default_trajectory[0].positions), 2)
+            self.assertGreater(len(bad_trajectory[0].positions), 2)
```

Comment:
Move these two lines out of the `with:` if possible. I think that should still work.